monkey/infection_monkey/exploit/zerologon.py

```diff
@@ -0,0 +1,740 @@
+"""
+Zerologon, CVE-2020-1472
+Implementation based on https://github.com/dirkjanm/CVE-2020-1472/ and https://github.com/risksense/zerologon/.
+"""
+
+from __future__ import division, print_function
+
+import cmd
+import io
+import logging
+import ntpath
+import os
+import sys
+import time
+import traceback
+from binascii import unhexlify
+
+import impacket
+from impacket.dcerpc.v5 import epm, nrpc, transport
+from impacket.dcerpc.v5.dcom import wmi
+from impacket.dcerpc.v5.dcomrt import DCOMConnection
+from impacket.dcerpc.v5.dtypes import NULL
+from impacket.examples.secretsdump import (LocalOperations, LSASecrets,
+                                           NTDSHashes, RemoteOperations,
+                                           SAMHashes)
+from impacket.smbconnection import SMBConnection
+
+from common.utils.exploit_enum import ExploitType
+from infection_monkey.exploit.HostExploiter import HostExploiter
+from infection_monkey.network.zerologon_fingerprint import ZerologonFinger
+
+LOG = logging.getLogger(__name__)
+
+
+_orig_stdout = None
+_new_stdout = None
+
+
+def _set_stdout_to_in_memory_text_stream():
```

Comment:
Why do this and not modify `RemoteShell` to biffer output?
When using this function, it an exception is thrown it won't be printed to stdout.

Reply:
We're using secretsdump which prints the hashes that we need (handled in a module we can't touch), so we're capturing the whole output by doing this and then extracting exactly what is needed.

Added a try/except block to handle exceptions.